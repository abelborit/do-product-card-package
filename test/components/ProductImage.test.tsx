import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { productElement2 } from '../data/products';

describe('ProductImage', () => {
  test('debe mostrar el componente correctamente con la imagen personalizada', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://img.jpg" title="Custom Tittle" />
    );

    // console.log(wrapper.toJSON());
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostrar el componente con la imagen del producto 2', () => {
    const wrapper = renderer.create(
      <ProductCard product={productElement2}>
        {() => (
          <ProductImage
            img={productElement2.img}
            title={productElement2.title}
          />
        )}
      </ProductCard>
    );

    // console.log(wrapper.toJSON());
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
