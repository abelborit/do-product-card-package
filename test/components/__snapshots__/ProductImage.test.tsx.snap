// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente con la imagen del producto 2 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Card con imagen"
    className="productImg undefined"
    src="/coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Custom Tittle"
  className="productImg undefined"
  src="https://img.jpg"
/>
`;
